app/views/items/_form.html.erb

```diff
@@ -0,0 +1,11 @@
+<div class="col-md-4">
+  <div class="form-group">
+    <%= form_for [@user, @user.items.build], remote: true do |f| %>
```

Comment:
the `remote: true` option suggests that this form is supposed to be able to submit asynchronously to the server (meaning it will use an AJAX call).  That can make for a nice user experience since there's no page reload.  However, it doesn't look like your create action in the controller is set up to respond to an AJAX call.  I think that's causing your problem.